### PR TITLE
Support private cas

### PR DESCRIPTION
**Pull request by QuentinBisson**
### What this PR does / why we need it

Closes https://github.com/giantswarm/roadmap/issues/3761 

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [ ] Follow deployment test procedure in the tests/manual_e2e directory and have a working branch.

### CODE REVIEW

File: pkg/resource/logging-config/alloy-logging-config.go [hunk 1]
```diff
@@ -46,8 +46,9 @@ func GenerateAlloyLoggingConfig(lc loggedcluster.Interface, observabilityBundleV
 		DefaultWorkloadClusterNamespaces []string
 		IsWorkloadCluster                bool
 		SecretName                       string
-		SupportPodLogs                   bool
-		SupportVPA                       bool
+
```
Comment (QuentinBisson) [inline pkg/resource/logging-config/alloy-logging-config.go:49]:
```suggestion
```

**Comment by QuentinBisson** [comment]
Disable reporting:
image

**Comment by QuentinBisson** [comment]
Tested on golem

File: pkg/resource/logging-config/alloy/logging.alloy.template [hunk 2]
```diff
@@ -1,11 +1,11 @@
 // Kubernetes pods logs
 {{- if .SupportPodLogs }}
 loki.source.podlogs "kubernetes_pods" {
-  forward_to      = [loki.relabel.kubernetes_pods.receiver]
```
Comment (hervenicol) [inline pkg/resource/logging-config/alloy/logging.alloy.template:4]: It's funny how some indentation changed in multiple places.
Is it a change with `gofmt` behaviour? Depending on `go` version?
We should make sure this is legit and a one-time change.

File: pkg/resource/logging-config/alloy/test/logging-config.alloy.170_WC_default_namespaces_empty.yaml [hunk 3]
```diff
@@ -328,3 +332,6 @@ alloy:
           path: /run/log/journal
       - name: alloy-tmp
         emptyDir: {}
+verticalPodAutoscaler:
```
Comment (hervenicol) [inline pkg/resource/logging-config/alloy/test/logging-config.alloy.170_WC_default_namespaces_empty.yaml:335]: This looks related to https://github.com/giantswarm/logging-operator/pull/264
But how did the tests pass if the test golden files were wrong? Maybe the tests don't work and always pass?

Comment (QuentinBisson) [reply pkg/resource/logging-config/alloy/test/logging-config.alloy.170_WC_default_namespaces_empty.yaml:335]: I think we're not running tests at all but I'm not sure

File: pkg/resource/logging-config/alloy/logging.alloy.template (hunk 2 above)
Comment (QuentinBisson) [reply pkg/resource/logging-config/alloy/logging.alloy.template:4]: No it's not, the issue is we were using both tabs and spaces in the same template

Comment (QuentinBisson) [reply pkg/resource/logging-config/alloy/logging.alloy.template:4]: I just made sure we use the same everywhere on this file

File: pkg/resource/logging-config/alloy/test/logging-config.alloy.170_WC_default_namespaces_empty.yaml (hunk 3 above)
Comment (QuentinBisson) [reply pkg/resource/logging-config/alloy/test/logging-config.alloy.170_WC_default_namespaces_empty.yaml:335]: I did run them myself so I know they pass but we probably want to add them to our pipeline

File: pkg/resource/logging-config/alloy/logging.alloy.template (hunk 2 above)
Comment (hervenicol) [reply pkg/resource/logging-config/alloy/logging.alloy.template:4]: Oh, it's not a go file, that's why!
`gofmt` does not run on this.

File: pkg/resource/logging-config/alloy/test/logging-config.alloy.170_WC_default_namespaces_empty.yaml (hunk 3 above)
Comment (hervenicol) [reply pkg/resource/logging-config/alloy/test/logging-config.alloy.170_WC_default_namespaces_empty.yaml:335]: Unit tests were never failing, fixed here: https://github.com/giantswarm/logging-operator/pull/267

**Review by hervenicol — APPROVED** [review]
LGTM

**Pull request merged**